test/JDBC/input/rowcount-vu-prepare.sql

```diff
@@ -1,46 +1,104 @@
     set rowcount 0;
     set rowcount 5;
-    insert into testing1 select a from testing1;
-    insert into testing1 select a from testing1;
+    insert into rowcount_vu_prepare_testing1 select a from rowcount_vu_prepare_testing1;
+    insert into rowcount_vu_prepare_testing1 select a from rowcount_vu_prepare_testing1;
     set rowcount 0;
 end
 GO
 
-create procedure rowcountselect as
+create procedure rowcount_vu_prepare_select_proc as
 begin
     set rowcount 1;
-    select count(*) from testing1;
-    select * from testing1;
+    select count(*) from rowcount_vu_prepare_testing1;
+    select * from rowcount_vu_prepare_testing1;
     set rowcount 0;
 end
 GO
 
-create procedure rowcountupdate as
+create procedure rowcount_vu_prepare_update_proc as
 begin
     set rowcount 1;
-    select count(*) from testing1;
-    update testing1 set a = 10;
-    select count(*) from testing1 where a = 10;
+    select count(*) from rowcount_vu_prepare_testing1;
+    update rowcount_vu_prepare_testing1 set a = 10;
+    select count(*) from rowcount_vu_prepare_testing1 where a = 10;
     set rowcount 0;
 end
 GO
 
-create procedure rowcountdelete as
+create procedure rowcount_vu_prepare_delete_proc as
 begin
     set rowcount 1;
-    select count(*) from testing1 where a = 1;
-    delete from testing1 where a = 1;
-    select count(*) from testing1 where a = 1;
+    select count(*) from rowcount_vu_prepare_testing1 where a = 1;
+    delete from rowcount_vu_prepare_testing1 where a = 1;
+    select count(*) from rowcount_vu_prepare_testing1 where a = 1;
+    set rowcount 0;
+end
+GO
+
+-- procedures to test "SET ROWCOUNT @variable"
+create procedure rowcount_vu_prepare_insert_proc_var as
+begin
+    declare @v int = 1;
+    set rowcount @v;
+    insert into rowcount_vu_prepare_testing2 values (1);
+    insert into rowcount_vu_prepare_testing2 values (1);
+    insert into rowcount_vu_prepare_testing2 select a + 1 from rowcount_vu_prepare_testing2;
+    insert into rowcount_vu_prepare_testing2 select a + 1 from rowcount_vu_prepare_testing2;
+    declare @x int = 0;
+    set rowcount @x;
+    set @x = 5;
+    set rowcount @x;
+    insert into rowcount_vu_prepare_testing2 select a from rowcount_vu_prepare_testing2;
+    insert into rowcount_vu_prepare_testing2 select a from rowcount_vu_prepare_testing2;
+    set rowcount 0;
+end
+GO
+
+create procedure rowcount_vu_prepare_select_proc_var as
+begin
+    declare @v smallint = 1;
+    set rowcount @v;
+    select count(*) from rowcount_vu_prepare_testing2;
+    select * from rowcount_vu_prepare_testing2;
+    declare @x int = 0;
+    set rowcount @x;
+end
+GO
+
+create procedure rowcount_vu_prepare_update_proc_var as
+begin
+    declare @v smallint = 1;
+    set rowcount @v;
+    select count(*) from rowcount_vu_prepare_testing2;
+    update rowcount_vu_prepare_testing2 set a = 10;
+    select count(*) from rowcount_vu_prepare_testing2 where a = 10;
+    declare @x int = 0;
+    set rowcount @x;
```

Comment:
Add a test case where SET ROWCOUNT is set in proc A which calls proc B which performs a SELECT and show the rowcount applies there.
Also add test cases with a loop which read/update/delete rows from a table one by one.